tests/foreman/api/test_docker.py

```diff
@@ -0,0 +1,680 @@
+            organization=self.org_id, composite=False).create_json()
+        _add_repo_to_content_view(repo_id, content_view['id'])
+
+        new_attrs = entities.ContentView(id=content_view['id']).read_json()
+        self.assertIn(repo_id, new_attrs['repository_ids'])
+        # Not published yet?
+        self.assertIsNone(new_attrs['last_published'])
+        self.assertEqual(new_attrs['next_version'], 1)
+
+        # Publish it...
+        self.assertTrue(_publish_content_view(content_view['id']))
+        # ... and check that it was indeed published
+        new_attrs = entities.ContentView(id=content_view['id']).read_json()
+        self.assertIsNotNone(new_attrs['last_published'])
+        self.assertGreater(new_attrs['next_version'], 1)
+
+    @run_only_on('sat')
+    def test_publish_once_docker_repo_composite_content_view(self):
+        """@Test: Add Docker-type repository to composite
+        content view and publish it once.
+
+        @Assert: One repository is created with a Docker image and the product
+        is added to a content view which is then published only once and then
+        added to a composite content view which is also published only once.
+
+        @Feature: Docker
+
+        """
+        prod_id = entities.Product(
+            organization=self.org_id
+        ).create_json()['id']
+
+        repo_id = _create_repository(prod_id)['id']
+
+        content_view = entities.ContentView(
+            organization=self.org_id, composite=False).create_json()
+        _add_repo_to_content_view(repo_id, content_view['id'])
+
+        new_attrs = entities.ContentView(id=content_view['id']).read_json()
+        self.assertIn(repo_id, new_attrs['repository_ids'])
+        # Not published yet?
+        self.assertIsNone(new_attrs['last_published'])
+        self.assertEqual(new_attrs['next_version'], 1)
+
+        # Publish it...
+        self.assertTrue(_publish_content_view(content_view['id']))
+        # ... and check that it was indeed published
+        new_attrs = entities.ContentView(id=content_view['id']).read_json()
+        version_id = new_attrs['versions'][0]['id']
+        self.assertIsNotNone(new_attrs['last_published'])
+        self.assertGreater(new_attrs['next_version'], 1)
+
+        # Create composite content view...
+        comp_content_view = entities.ContentView(
+            organization=self.org_id, composite=True).create_json()
+        # ... add content view to it...
+        _add_content_view_to_composite_view(
+            comp_content_view['id'], version_id)
+        new_attrs = entities.ContentView(
+            id=comp_content_view['id']).read_json()
+        self.assertIn(version_id, new_attrs['component_ids'])
+        # ... publish it...
+        self.assertTrue(_publish_content_view(comp_content_view['id']))
+        # ... and check that it was indeed published
+        new_attrs = entities.ContentView(
+            id=comp_content_view['id']).read_json()
```

Comment:
`id` is the only attribute accessed in `comp_content_view`. Can you do something like this?

    comp_content_view_id = entities.ContentView(
        ...
    ).create_json()['id']

You can even go a step further and do this:

    comp_content_view = entities.ContentView(id=comp_content_view_id)
    ...
    new_attrs = comp_content_view.read_json()